pytradfri/smart_task.py

```diff
@@ -143,80 +194,104 @@ def __repr__(self):
 class TaskControl:
     """Class to control the tasks."""
 
-    def __init__(self, task, state, path, gateway):
+    def __init__(
+        self, task: SmartTask, state: bool, path: list[str], gateway: Gateway
+    ) -> None:
         """Initialize TaskControl."""
         self._task = task
         self.state = state
         self.path = path
         self._gateway = gateway
 
     @property
-    def tasks(self):
+    def tasks(self) -> list[StartActionItem]:
         """Return task objects of the task control."""
         return [
-            StartActionItem(self._task, i, self.state, self.path, self.raw)
-            for i in range(len(self.raw))
+            StartActionItem(self._task, idx, self.state, self.path, self.raw)
+            for idx in range(len(self.raw.root_start_action))
         ]
 
-    def set_dimmer_start_time(self, hour, minute):
+    def calibrate_time(self) -> Command[None]:
+        """Calibrate difference between local time and gateway time."""
+
+        def process_result(result: TypeRaw) -> None:
+            gateway_info: GatewayInfo = GatewayInfo(result)
+            if gateway_info.current_time:
```

Comment:
I'd invert this check and `return` if it's true. Then we can outdent below.

Reply:
Sorry, realise now that didn't quite understand your comment for this yesterday. Is this what you meant?

Reply:
Yes. 👍